.github/workflows/cmake.yml

```diff
@@ -12,26 +12,31 @@ env:
 
 jobs:
   build:
+    if: "github.repository == 'DataMedSci/gi-yagit' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]')"
+
     # The CMake configure and build commands are platform agnostic and should work equally
     # well on Windows or Mac.  You can convert this to a matrix build if you need
     # cross-platform coverage.
     # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest]
+        
     steps:
     - uses: actions/checkout@v2
 
     - name: Configure CMake
       # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
       # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
-      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+      run: cd gi_core && cmake -B ${{github.workspace}}/gi_core/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
 
     - name: Build
       # Build your program with the given configuration
-      run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+      run: cd gi_core && cmake --build ${{github.workspace}}/gi_core/build --config ${{env.BUILD_TYPE}}
```

Comment:
mam już gotowy plik cmake dla wszystkich podprojektów - na branchu cmake_support, za niedługo postaram się go zintegrować z tym CI (windows jest lekkim problemem bo musiałbym dociągnąć wxWidgets za pomocą vcpkg), ale generalnie jak na ten moment to chyba niech tak zostanie